Eevee.Sleep.Bot/Workers/InGameAnnouncementCrawlingWorker.cs

```diff
@@ -0,0 +1,86 @@
+using AngleSharp.Common;
+using Discord.WebSocket;
+using Eevee.Sleep.Bot.Controllers.Mongo;
+using Eevee.Sleep.Bot.Enums;
+using Eevee.Sleep.Bot.Exceptions;
+using Eevee.Sleep.Bot.Extensions;
+using Eevee.Sleep.Bot.Models;
+using Eevee.Sleep.Bot.Utils;
+using Eevee.Sleep.Bot.Workers.Scrapers.InGameAnnouncement;
+
+namespace Eevee.Sleep.Bot.Workers;
+public class InGameAnnouncementCrawlingWorker(
+        DiscordSocketClient client,
+        ILogger<InGameAnnouncementCrawlingWorker> logger
+    ) : BackgroundService {
+    private readonly DiscordSocketClient _client = client;
+    private readonly ILogger<InGameAnnouncementCrawlingWorker> _logger = logger;
+    private readonly CancellationTokenSource _cancellationTokenSource = new();
+
+    private static readonly TimeSpan CheckInterval = TimeSpan.FromSeconds(60);
+
+    protected override async Task ExecuteAsync(CancellationToken cancellationToken) {
+        _logger.LogInformation("InGameAnnouncementUpdateCheckWorker is starting.");
+        cancellationToken.Register(() => _logger.LogInformation("InGameAnnouncementUpdateCheckWorker background task is stopping."));
+
+        while (!cancellationToken.IsCancellationRequested) {
+            _logger.LogInformation("InGameAnnouncementUpdateCheckWorker task doing background work.");
+
+            try {
+                var indexes = await GetIndexes();
+                await InGameAnnouncememntIndexController.BulkUpsert([.. indexes]);
+                
+                var details = await GetDetails(indexes);
+                await SaveDetailsAndHistories(details);
+            } catch (ContentStructureChangedException e) {
+                _logger.LogError("Failed to get indexes. Web page structure may have changed.");
+                await _client.SendMessageInAdminAlertChannel(
+                    embed: DiscordMessageMaker.MakeContentStructureChangedMessage(e)
+                );
+                _cancellationTokenSource.Cancel();
+                break;
+            }
+
+            await Task.Delay(CheckInterval, cancellationToken);
+        }
+    }
+
+    private static async Task<IEnumerable<InGameAnnouncementIndexModel>> GetIndexes() {
+        Dictionary<string, InGameAnnoucementLanguages> urls = new(){
```

Comment:
Expand these to crawl every pages, so it won't be only `page/2`.

Since the page bottom has an option list of all pages, I would assume that it's possible to know how many pages are there for each language

![image](https://github.com/RaenonX-PokemonSleep/eevee-sleep-discord-bot/assets/18443743/012ded38-6389-4ccb-9d66-87bc2de768fc)